File: Javascript/check.js
/**
 * Created by isee15 on 15/1/15.
 */
var http = require('http');
var net = require('net');
var url = require('url');
//var querystring = require('querystring');
var lsconverter = require('./LunarSolarConverter.io.js');

// Create an HTTP server
var proxy = http.createServer(function (req, res) {
    var url_parts = url.parse(req.url, true);
    var query = url_parts.query;
    res.writeHead(200, {'Content-Type': 'text/plain'});
    var converter = new lsconverter.LunarSolarConverter();
    var queryParam = String(query['src']);
    var params = queryParam.split(",");
    console.log(params);
    var contents = queryParam + " format is not correct,try http://localhost:1337/?src=2015,1,15 or http://localhost:1337/?src=2015,1,15,0";
    if (params.length == 3) {
        var solar = new lsconverter.Solar();
        solar.solarYear = parseInt(params[0]);
        solar.solarMonth = parseInt(params[1]);
        solar.solarDay = parseInt(params[2]);
        var lunar = converter.SolarToLunar(solar);
        console.log(lunar);
        contents = "" + lunar.lunarYear + "," + lunar.lunarMonth + "," + lunar.lunarDay + "," + (lunar.isleap ? 1 : 0) + "";
    } else if (params.length == 4) {
        var lunar = new lsconverter.Lunar();
        lunar.lunarYear = parseInt(params[0]);
        lunar.lunarMonth = parseInt(params[1]);
        lunar.lunarDay = parseInt(params[2]);
        lunar.isleap = (params[3] == 1);
        var solar = converter.LunarToSolar(lunar);
        console.log(solar);
        contents = "" + solar.solarYear + "," + solar.solarMonth + "," + solar.solarDay + "";
    }
    console.log(contents);
    res.end(contents);
});

// now that proxy is running
proxy.listen(1337, '127.0.0.1', function () {

});

File: C#/Check.cs
using System;
using System.Globalization;
using System.IO;
using System.Net;
using System.Text;
using Cn.Z;

namespace Util
{
    internal class Check
    {
        private static readonly ChineseLunisolarCalendar cnCalendar = new ChineseLunisolarCalendar();
        private static string GetResult(string src)
        {
            try
            {
                string url = "http://localhost:1337/?src=" + src;
                System.Net.WebRequest wReq = System.Net.WebRequest.Create(url);
                // Get the response instance.
                System.Net.WebResponse wResp = wReq.GetResponse();
                System.IO.Stream respStream = wResp.GetResponseStream();
                using (System.IO.StreamReader reader = new System.IO.StreamReader(respStream, Encoding.GetEncoding("UTF-8")))
                {
                    return reader.ReadToEnd();
                }
            }
            catch (System.Exception ex)
            {
                //errorMsg = ex.Message;
            }
            return "";
        }
        public static void Verify()
        {
            DateTime endDt = cnCalendar.MaxSupportedDateTime;
            for (DateTime dt = cnCalendar.MinSupportedDateTime; dt < endDt; dt = dt.AddDays(1))
            {
                Solar sd = new Solar();
                sd.solarYear = dt.Year;
                sd.solarMonth = dt.Month;
                sd.solarDay = dt.Day;
                Lunar ld = LunarSolarConverter.SolarToLunar(sd);
                string solarString = dt.Year + "," + dt.Month + "," + dt.Day;
                string lunarString = GetResult(solarString);
                string solar2String = GetResult(lunarString);
                if (lunarString != (ld.lunarYear + "," + ld.lunarMonth + "," + ld.lunarDay + "," + (ld.isleap ? 1 : 0)))
                {
                    Console.WriteLine("lunar error:" + lunarString);
                }
                if (solarString != solar2String)
                {
                    Console.WriteLine("solar error:" + solarString);
                }
                
            }
        }

        private static void Main(string[] args)
        {
            Verify();
            Console.WriteLine("Verify Done---------------------------------------");
            Console.Read();
        }
    }
}